tests/bsim/bluetooth/audio/src/cap_commander_test.c

```diff
@@ -195,6 +244,398 @@ static void cap_disconnected_cb(struct bt_conn *conn, uint8_t reason)
+	if (ret != 0) {
+		FAIL("Failed to BIS indexes: %d\n", ret);
+		return;
+	}
+
+	bis_index_bitfield = base_bis_index_bitfield & bis_index_mask;
+
+	SET_FLAG(flag_base_received);
+}
+
+static void syncable_cb(struct bt_bap_broadcast_sink *sink, const struct bt_iso_biginfo *biginfo)
+{
+	printk("Broadcast sink %p syncable with%s encryption\n", sink,
+	       biginfo->encryption ? "" : "out");
+	SET_FLAG(flag_syncable);
+}
+
+static struct bt_bap_broadcast_sink_cb broadcast_sink_cbs = {
+	.base_recv = base_recv_cb,
+	.syncable = syncable_cb,
+};
+
+static void bap_broadcast_assistant_discover_cb(struct bt_conn *conn, int err,
+						uint8_t recv_state_count)
+{
+	if (err == 0) {
+		printk("BASS discover done with %u recv states\n", recv_state_count);
+	} else {
+		printk("BASS discover failed (%d)\n", err);
+	}
+	k_sem_give(&sem_ba_discovered);
+}
+
+static void bap_broadcast_assistant_add_src_cb(struct bt_conn *conn, int err)
+{
+	if (err == 0) {
+		printk("BASS add source successful\n");
+	} else {
+		printk("BASS add source failed (%d)\n", err);
+	}
+}
+
+static bool metadata_entry(struct bt_data *data, void *user_data)
+{
+	char metadata[512];
+
+	(void)bin2hex(data->data, data->data_len, metadata, sizeof(metadata));
+
+	printk("\t\tMetadata length %u, type %u, data: %s\n", data->data_len, data->type, metadata);
+
+	return true;
+}
+
+static void
+bap_broadcast_assistant_recv_state_cb(struct bt_conn *conn, int err,
+				      const struct bt_bap_scan_delegator_recv_state *state)
+{
+	char le_addr[BT_ADDR_LE_STR_LEN];
+	char bad_code[33];
+
+	if (err != 0) {
+		FAIL("BASS recv state read failed (%d)\n", err);
+		return;
+	}
+
+	if (state == NULL) {
+		/* Empty receive state */
+		return;
+	}
+
+	bt_addr_le_to_str(&state->addr, le_addr, sizeof(le_addr));
+	(void)bin2hex(state->bad_code, BT_AUDIO_BROADCAST_CODE_SIZE, bad_code, sizeof(bad_code));
+	printk("BASS recv state: src_id %u, addr %s, sid %u, sync_state %u, "
+	       "encrypt_state %u%s%s\n",
+	       state->src_id, le_addr, state->adv_sid, state->pa_sync_state, state->encrypt_state,
+	       state->encrypt_state == BT_BAP_BIG_ENC_STATE_BAD_CODE ? ", bad code" : "", bad_code);
+
+	for (int i = 0; i < state->num_subgroups; i++) {
+		const struct bt_bap_bass_subgroup *subgroup = &state->subgroups[i];
+		struct net_buf_simple buf;
+
+		printk("\t[%d]: BIS sync %u, metadata_len %u\n", i, subgroup->bis_sync,
+		       subgroup->metadata_len);
+
+		net_buf_simple_init_with_data(&buf, (void *)subgroup->metadata,
+					      subgroup->metadata_len);
+		bt_data_parse(&buf, metadata_entry, NULL);
+
+		if (subgroup->bis_sync != 0) {
+			SET_FLAG(flag_recv_state_updated_with_bis_sync);
+		}
+	}
+
+	if (state->pa_sync_state == BT_BAP_PA_STATE_INFO_REQ) {
+		err = bt_le_per_adv_sync_transfer(g_pa_sync, conn, BT_UUID_BASS_VAL);
+		if (err != 0) {
+			FAIL("Could not transfer periodic adv sync: %d\n", err);
+			return;
+		}
+	}
+
+	if (state->pa_sync_state == BT_BAP_PA_STATE_SYNCED) {
+	}
```

Comment:
What is the intention here?

Reply:
keeping it so that original and copy stay in sync, plan is to move these into a common section